src/Storages/ObjectStorage/DataLakes/IStorageDataLake.h

```diff
@@ -102,12 +115,28 @@ class IStorageDataLake final : public StorageObjectStorage
         Storage::updateConfiguration(local_context);
 
         auto new_metadata = DataLakeMetadata::create(Storage::object_storage, base_configuration, local_context);
-        if (current_metadata && *current_metadata == *new_metadata)
-            return;
+        if (!current_metadata || (*current_metadata != *new_metadata))
+        {
+            if constexpr (std::is_same_v<IcebergMetadata, DataLakeMetadata>)
+            {
+                throw Exception(
+                    ErrorCodes::FORMAT_VERSION_TOO_OLD,
+                    "Storage thinks that actual metadata version is {}, but actual metadata version is {} current",
+                    (dynamic_cast<IcebergMetadata *>(current_metadata.get()) != nullptr)
+                        ? dynamic_cast<IcebergMetadata *>(current_metadata.get())->getVersion()
+                        : -1,
+                    (dynamic_cast<IcebergMetadata *>(new_metadata.get()) != nullptr)
+                        ? dynamic_cast<IcebergMetadata *>(new_metadata.get())->getVersion()
+                        : -1);
+            }
+            else
+            {
+                current_metadata = std::move(new_metadata);
+            }
+        }
```

Comment:
We can add virtual method `getVersion()` to `IDataLakeMetadata`

Reply:
It is unclear how to do this function for DeltaLake